### ISSUE TITLE
LimLog: inner::test_map failed on Windows

### ISSUE THREAD

**Issue by ho-229** [Limit-LAB/limlog#18]
OS: Windows 10 21H2 LTSC
Rust: rustc 1.70.0-nightly (f63ccaf25 2023-03-06)

```txt
running 3 tests
test util::test_subarray ... ok
test formats::log::test_log_size ... ok
test inner::test_map ... FAILED

failures:

---- inner::test_map stdout ----
thread 'inner::test_map' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Os { code: 1224, kind: Uncategorized, message: "请求的操作无法在使用用户映射区域打开的文件上执行。" })', src\inner.rs:150:44
stack backtrace:
   0:     0x7ff70face3f2 - std::backtrace_rs::backtrace::dbghelp::trace
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
   1:     0x7ff70face3f2 - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff70face3f2 - std::sys_common::backtrace::_print_fmt
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:65
   3:     0x7ff70face3f2 - std::sys_common::backtrace::_print::impl$0::fmt
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:44
   4:     0x7ff70faee3db - core::fmt::write
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\fmt\mod.rs:1232
   5:     0x7ff70fac9a6a - std::io::Write::write_fmt<alloc::vec::Vec<u8,alloc::alloc::Global> >
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\io\mod.rs:1684
   6:     0x7ff70face13b - std::sys_common::backtrace::_print
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:47
   7:     0x7ff70face13b - std::sys_common::backtrace::print
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:34
   8:     0x7ff70fad1229 - std::panicking::default_hook::closure$1
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:271
   9:     0x7ff70fad0e33 - std::panicking::default_hook
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:287
  10:     0x7ff70fa653d8 - alloc::boxed::impl$47::call
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\alloc\src\boxed.rs:2001
  11:     0x7ff70fa653d8 - test::test_main::closure$0
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:135 
  12:     0x7ff70fad1af2 - alloc::boxed::impl$47::call
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\alloc\src\boxed.rs:2001
  13:     0x7ff70fad1af2 - std::panicking::rust_panic_with_hook
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:696
  14:     0x7ff70fad184e - std::panicking::begin_panic_handler::closure$0
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:583
  15:     0x7ff70facf0b9 - std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:150
  16:     0x7ff70fad1500 - std::panicking::begin_panic_handler
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:579
  17:     0x7ff70faf82e5 - core::panicking::panic_fmt
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\panicking.rs:64
  18:     0x7ff70faf8796 - core::result::unwrap_failed
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\result.rs:1750
  19:     0x7ff70fa17216 - enum2$<core::result::Result<tuple$<>,enum2$<limlog::error::ErrorType> > >::unwrap<tuple$<>,enum2$<limlog::error::ErrorType> >
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2\library\core\src\result.rs:1090
  20:     0x7ff70fa1ac64 - limlog::inner::impl$2::drop
                               at C:\Projects\rust\limlog\src\inner.rs:150
  21:     0x7ff70fa1aa7f - core::ptr::drop_in_place<limlog::inner::SharedMap>
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2\library\core\src\ptr\mod.rs:490
  22:     0x7ff70fa1240e - limlog::inner::test_map
                               at C:\Projects\rust\limlog\src\inner.rs:282
  23:     0x7ff70fa13788 - limlog::inner::test_map::closure$0
                               at C:\Projects\rust\limlog\src\inner.rs:255
  24:     0x7ff70fa1a812 - core::ops::function::FnOnce::call_once<limlog::inner::test_map::closure_env$0,tuple$<> >
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2\library\core\src\ops\function.rs:250
  25:     0x7ff70fa6c1bf - core::ops::function::FnOnce::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\ops\function.rs:250
  26:     0x7ff70fa6c1bf - test::__rust_begin_short_backtrace<enum2$<core::result::Result<tuple$<>,alloc::string::String> >,enum2$<core::result::Result<tuple$<>,alloc::string::String> > (*)()>
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:656 
  27:     0x7ff70fa34e80 - test::run_test::closure$1
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:647 
  28:     0x7ff70fa34e80 - core::ops::function::FnOnce::call_once<test::run_test::closure_env$1,tuple$<> >    
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\ops\function.rs:250
  29:     0x7ff70fa6a9f9 - alloc::boxed::impl$45::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\alloc\src\boxed.rs:1987
  30:     0x7ff70fa6a9f9 - core::panic::unwind_safe::impl$23::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\panic\unwind_safe.rs:271
  31:     0x7ff70fa6a9f9 - std::panicking::try::do_call
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:487
  32:     0x7ff70fa6a9f9 - std::panicking::try
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:451
  33:     0x7ff70fa6a9f9 - std::panic::catch_unwind
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panic.rs:140  34:     0x7ff70fa6a9f9 - test::run_test_in_process
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:679 
  35:     0x7ff70fa6a9f9 - test::run_test::run_test_inner::closure$0
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:573 
  36:     0x7ff70fa2bd8a - test::run_test::run_test_inner::closure$1
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\test\src\lib.rs:600 
  37:     0x7ff70fa2bd8a - std::sys_common::backtrace::__rust_begin_short_backtrace<test::run_test::run_test_inner::closure_env$1,tuple$<> >
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys_common\backtrace.rs:134
  38:     0x7ff70fa34b5b - std::thread::impl$0::spawn_unchecked_::closure$1::closure$0
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\thread\mod.rs:560
  39:     0x7ff70fa34b5b - core::panic::unwind_safe::impl$23::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\panic\unwind_safe.rs:271
  40:     0x7ff70fa34b5b - std::panicking::try::do_call
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:487
  41:     0x7ff70fa34b5b - std::panicking::try
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panicking.rs:451
  42:     0x7ff70fa34b5b - std::panic::catch_unwind
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\panic.rs:140  43:     0x7ff70fa34b5b - std::thread::impl$0::spawn_unchecked_::closure$1
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\thread\mod.rs:559
  44:     0x7ff70fa34b5b - core::ops::function::FnOnce::call_once<std::thread::impl$0::spawn_unchecked_::closure_env$1<test::run_test::run_test_inner::closure_env$1,tuple$<> >,tuple$<> >
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\core\src\ops\function.rs:250
  45:     0x7ff70faddbfc - alloc::boxed::impl$45::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\alloc\src\boxed.rs:1987
  46:     0x7ff70faddbfc - alloc::boxed::impl$45::call_once
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\alloc\src\boxed.rs:1987
  47:     0x7ff70faddbfc - std::sys::windows::thread::impl$0::new::thread_start
                               at /rustc/f63ccaf25f74151a5d8ce057904cd944074b01d2/library\std\src\sys\windows\thread.rs:56
  48:     0x7ffe4d077614 - BaseThreadInitThunk
  49:     0x7ffe4dc626a1 - RtlUserThreadStart


failures:
    inner::test_map

test result: FAILED. 2 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.08s

error: test failed, to rerun pass `--lib`
```

**Comment by ho-229** [comment]
updated

**Comment by ho-229** [comment]
Fix in [c9fae4e](https://github.com/Limit-LAB/limlog/commit/c9fae4ebcc2f6036be3f6e9aeceddce955b3e043)
